pkg/client/cache/reflector.go

```diff
@@ -232,12 +233,20 @@ func (r *Reflector) listAndWatch(stopCh <-chan struct{}) {
 }
 
 // syncWith replaces the store's items with the given list.
-func (r *Reflector) syncWith(items []runtime.Object) error {
+func (r *Reflector) syncWith(items []runtime.Object, resourceVersion string) error {
 	found := make([]interface{}, 0, len(items))
 	for _, item := range items {
 		found = append(found, item)
 	}
 
+	myStore, ok := r.store.(*WatchCache)
```

Comment:
It's weird to do the string->int conversion here.  Would suggest this become an interface VersionedStore that implements ReplaceWithVersion(string).  Reflector shouldn't have to know about int resource versions.